agent/src/agentclient.rs

```diff
@@ -257,45 +257,133 @@ impl<T: APIServerClient> BrupopAgent<T> {
         Ok(())
     }
 
+    #[instrument(skip(self))]
+    async fn create_custom_resource_if_not_exist(&self) -> Result<()> {
+        let custom_resource_exists = self.check_node_custom_resource_exists().await?;
+        if !custom_resource_exists {
+            self.create_metadata_custom_resource().await?;
+        }
+        Ok(())
+    }
+
+    #[instrument(skip(self))]
+    async fn initialize_custom_resource_if_not_initialzed(&self) -> Result<()> {
```

Comment:
nit: typo, plus suggestion to improve clarity.

```suggestion
    async fn initialize_shadow_if_not_initialized(&self) -> Result<()> {
```

Reply:
I wonder if it generally makes sense to refer to the `custom_resource` as the `shadow` in the agent. We avoided it at first because `bottlerocket_node` was ambiguous, but now we have an unambiguous name to refer to.